OpenRA.Game/Support/LaunchArguments.cs

```diff
@@ -38,17 +41,20 @@ public LaunchArguments(Arguments args)
 					FieldLoader.LoadField(this, f.Name, args.GetValue("Launch" + "." + f.Name, ""));
 		}
 
-		public string GetConnectAddress()
+		public ConnectionTarget GetConnectEndPoint()
 		{
-			var connect = string.Empty;
-
-			if (!string.IsNullOrEmpty(Connect))
-				connect = Connect;
-
+			Uri uri;
 			if (!string.IsNullOrEmpty(URI))
-				connect = URI.Substring(URI.IndexOf("://", System.StringComparison.Ordinal) + 3).TrimEnd('/');
-
-			return connect;
+				uri = new Uri(URI);
+			else if (!string.IsNullOrEmpty(Connect))
+				uri = new Uri("tcp://" + Connect);
+			else
```

Comment:
Style nits: This could simplify slightly to
```csharp
Uri uri = null;
if (!string.IsNullOrEmpty(URI))
	uri = new Uri(URI);
else if (!string.IsNullOrEmpty(Connect))
	uri = new Uri("tcp://" + Connect);

if (uri != null && uri.IsAbsoluteUri)
	return new ConnectionTarget(uri.Host, uri.Port);

return null;
```

Reply:
Both the current code and my suggestion fail to filter out bogus addresses:
* `Launch.URI=fail` crashes at the Uri constructor.
* `Launch.Connect=fail` is parsed as a valid address, only to crash deep in the connection code when we try to use it.

The expected behaviour for both cases is to `return null` and load to the shellmap ignoring the argument.

Reply:
Not sure about the second crash, the Uri constructor rejects it in my tests. But maybe another change already solved that.